### PR TITLE
aws[minor]: Implement WSO with tool_choice

### PR DESCRIPTION
**Pull request by bracesproul**

### CODE REVIEW

File: libs/langchain-aws/src/chat_models.ts [hunk 1]
```diff
@@ -430,4 +442,138 @@ export class ChatBedrockConverse
...
+  ):
+    | Runnable<BaseLanguageModelInput, RunOutput>
+    | Runnable<
+        BaseLanguageModelInput,
+        {
+          raw: BaseMessage;
+          parsed: RunOutput;
+        }
+      > {
+    if (typeof this.bindTools !== "function") {
```
Comment (jacoblee93) [inline libs/langchain-aws/src/chat_models.ts:487]: Think we can drop this here

File: libs/langchain-aws/src/chat_models.ts [hunk 2]
```diff
@@ -430,4 +442,138 @@ export class ChatBedrockConverse
...
+    }
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const schema: z.ZodType<RunOutput> | Record<string, any> = outputSchema;
+    const name = config?.name;
+    const description = schema.description ?? "A function available to call.";
+    const method = config?.method;
+    const includeRaw = config?.includeRaw;
+    if (method === "jsonMode") {
+      throw new Error(
+        `Base withStructuredOutput implementation only supports "functionCalling" as a method.`
```
Comment (jacoblee93) [inline libs/langchain-aws/src/chat_models.ts:500]: Drop

File: libs/langchain-aws/src/chat_models.ts [hunk 3]
```diff
@@ -430,4 +442,138 @@ export class ChatBedrockConverse
...
+            name: functionName,
+            description,
+            parameters: schema,
+          },
+        },
+      ];
+    }
+
+    const llm = this.bindTools(tools, {
+      tool_choice: tools[0].function.name,
```
[the remark this answers is not in the document]
Comment (jacoblee93) [reply libs/langchain-aws/src/chat_models.ts:534]: Should probably only do for Anthropic models?

Should we additionally add a check for non-Anthropic models in `bindTools`?

[the remark this answers is not in the document]
Comment (bracesproul) [reply libs/langchain-aws/src/chat_models.ts]: Updated to
- check if model supports tool_choice, and if so pass it when binding tools in WSO
- throw if passing tool_choice with an unsupported model when calling `.bindTools`
- added extensive tests verifying it throws/doesn't throw when expected, and `tool_choice` is/isn't bound to the model when expected/unexpected

**Pull request merged**